src/DotVVM.Framework/Resources/Scripts/utils/stringHelper.ts

```diff
@@ -49,3 +49,43 @@ export function join<T>(elements: T[], delimiter: string): string {
 export function format(pattern: string, expressions: any[]): string {
     return formatImpl(pattern, ...expressions);
 }
+
+export function trimStart(text: string, char: string) {
+    if (char != null) {
+        char = char.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+    }
+    else {
+        char = "\\s";
+    }
+    return text.replace(new RegExp("^" + char + "*"), "");
+}
+
+export function trimEnd(text: string, char: string) {
+    if (char != null) {
+        char = char.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+    }
+    else {
+        char = "\\s";
+    }
+    return text.replace(new RegExp(char + "*$"), "");
+}
+
+export function padStart(text: string, length: number, char: string) {
+    if (length <= text.length) {
```

Comment:
I think you can use this function: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart. We don't support IE anyway

Reply:
We cannot drop IE support at main. This will be possible for v4.